docs/automatic-installation.md

```diff
@@ -0,0 +1,119 @@
+- `JOIN`: we are installing a node to join an existing Harvester cluster.
+
+### CREATE mode
+
+Create a [Harvester configuration file](./harvester-configuration.md) `config-create.yaml` for `CREATE` mode, modify the values as needed:
+
+```
+# cat /usr/share/nginx/html/harvester/config-create.yaml
+token: token
+os:
+  hostname: node1
+  ssh_authorized_keys:
+  - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDbeUa9A7Kee+hcCleIXYxuaPksn2m4PZTd4T7wPcse8KbsQfttGRax6vxQXoPO6ehddqOb2nV7tkW2mEhR50OE7W7ngDHbzK2OneAyONYF44bmMsapNAGvnsBKe9rNrev1iVBwOjtmyVLhnLrJIX+2+3T3yauxdu+pmBsnD5OIKUrBrN1sdwW0rA2rHDiSnzXHNQM3m02aY6mlagdQ/Ovh96h05QFCHYxBc6oE/mIeFRaNifa4GU/oELn3a6HfbETeBQz+XOEN+IrLpnZO9riGyzsZroB/Y3Ju+cJxH06U0B7xwJCRmWZjuvfFQUP7RIJD1gRGZzmf3h8+F+oidkO2i5rbT57NaYSqkdVvR6RidVLWEzURZIGbtHjSPCi4kqD05ua8r/7CC0PvxQb1O5ILEdyJr2ZmzhF6VjjgmyrmSmt/yRq8MQtGQxyKXZhJqlPYho4d5SrHi5iGT2PvgDQaWch0I3ndEicaaPDZJHWBxVsCVAe44Wtj9g3LzXkyu3k= root@admin
+  password: rancher
+install:
+  mode: create
+  mgmt_interface: eth0
+  device: /dev/sda
+  iso_url: http://10.100.0.10/harvester-amd64.iso
+
+```
+
+For machines that needs to be installed as `CREATE` mode, the following is an iPXE script that boots the kernel with the above config:
+
+```
+#!ipxe
+kernel http://10.100.0.10/harvester/vmlinuz k3os.mode=install console=ttyS0 console=tty1 harvester.install.automatic=true harvester.install.config_url=http://10.100.0.10/harvester/config-create.yaml
+initrd http://10.100.0.10/harvester/initrd
+boot
+```
+
+### JOIN mode
+
+Create a [Harvester configuration file](./harvester-configuration.md) `config-join.yaml` for `JOIN` mode, modify the values as needed:
+
+```
+# cat /usr/share/nginx/html/harvester/config-join.yaml
+server_url: https://10.100.0.130:6443
+token: token
+os:
+  hostname: node2
+  ssh_authorized_keys:
+  - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDbeUa9A7Kee+hcCleIXYxuaPksn2m4PZTd4T7wPcse8KbsQfttGRax6vxQXoPO6ehddqOb2nV7tkW2mEhR50OE7W7ngDHbzK2OneAyONYF44bmMsapNAGvnsBKe9rNrev1iVBwOjtmyVLhnLrJIX+2+3T3yauxdu+pmBsnD5OIKUrBrN1sdwW0rA2rHDiSnzXHNQM3m02aY6mlagdQ/Ovh96h05QFCHYxBc6oE/mIeFRaNifa4GU/oELn3a6HfbETeBQz+XOEN+IrLpnZO9riGyzsZroB/Y3Ju+cJxH06U0B7xwJCRmWZjuvfFQUP7RIJD1gRGZzmf3h8+F+oidkO2i5rbT57NaYSqkdVvR6RidVLWEzURZIGbtHjSPCi4kqD05ua8r/7CC0PvxQb1O5ILEdyJr2ZmzhF6VjjgmyrmSmt/yRq8MQtGQxyKXZhJqlPYho4d5SrHi5iGT2PvgDQaWch0I3ndEicaaPDZJHWBxVsCVAe44Wtj9g3LzXkyu3k= root@admin
+  dns_nameservers:
+  - 1.1.1.1
+  - 8.8.8.8
+  password: rancher
+install:
+  mode: join
+  mgmt_interface: eth0
+  device: /dev/sda
+  iso_url: http://10.100.0.10/harvester/harvester-amd64.iso
+```
+
+Note the `mode` is `join` and `server_url` needs to be provided.
+
+For machines that needs to be installed as `JOIN` mode, the following is an iPXE script that boots the kernel with the above config:
+
+```
+#!ipxe
+kernel http://10.100.0.10/harvester/vmlinuz k3os.mode=install console=ttyS0 console=tty1 harvester.install.automatic=true harvester.install.config_url=http://10.100.0.10/harvester/config-join.yaml
```

Comment:
The same.

Reply:
This is still required for k3os to load harvester-installer.